js/hotgraphicView.js

```diff
@@ -282,6 +285,8 @@ define([
         },
 
         preRemove: function() {
+            if (!this.completionEvent) return;
```

Comment:
Well... That's not good. It suggests to me that there is a logic flaw in the removal of the view on the swap to a narrative. I'll have a look later.